src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3LinuxDriver.cs

```diff
@@ -350,6 +350,87 @@ private void SetInputPullModePi4(int pinNumber, PinMode mode)
             }
         }
 
+        /// <inheritdoc />
+        protected internal override void SetAlternatePinMode(int pinNumber, AlternatePinMode altMode)
+        {
+            Initialize();
+            /*
+             * There are 6 registers (4-byte ints) that control the mode for all pins. Each
+             * register controls the mode for 10 pins. Each pin uses 3 bits in the register
+             * containing the mode.
+             */
+
+            // Define the shift to get the right 3 bits in the register
+            int shift = (pinNumber % 10) * 3;
+            // Gets a pointer to the register that holds the mode for the pin
+            uint* registerPointer = &_registerViewPointer->GPFSEL[pinNumber / 10];
+            uint register = *registerPointer;
+            // Clear the 3 bits to 0 for the pin Number.
+            register &= ~(0b111U << shift);
+            // Set the 3 bits to the desired mode for that pin.
+            uint modeBits = 0; // Default: Gpio input
+
+            if (altMode == AlternatePinMode.Gpio)
+            {
+                // When setting back to Gpio, set to input, as this is the default
+                modeBits = 0;
+            }
+            else if (altMode.ModeValue < 8)
+            {
+                modeBits = altMode.ModeValue;
+            }
+            else
+            {
+                throw new InvalidOperationException($"Unknown Alternate pin mode value: {altMode.ModeValue}");
+            }
+
+            register |= (modeBits) << shift;
+            *registerPointer = register;
+        }
+
+        /// <inheritdoc />
+        protected internal override AlternatePinMode GetAlternatePinMode(int pinNumber)
+        {
+            Initialize();
+            /*
+             * There are 6 registers(4-byte ints) that control the mode for all pins. Each
+             * register controls the mode for 10 pins. Each pin uses 3 bits in the register
+             * containing the mode.
+             */
+
+            // Define the shift to get the right 3 bits in the register
+            int shift = (pinNumber % 10) * 3;
+            // Gets a pointer to the register that holds the mode for the pin
+            uint* registerPointer = &_registerViewPointer->GPFSEL[pinNumber / 10];

```

Comment:
make sure to throw here when pinNumber is gonna cause OutOfRange here. This is writing to memory directly so you should protected against that here